home/views.py

```diff
@@ -23,265 +19,258 @@ def index(request):
-            if not os.path.exists(user_workspace_dir):
-                os.makedirs(user_workspace_dir, exist_ok=True)
-
-            existing_workspaces = os.listdir(user_workspace_dir)
-            if existing_workspaces:
-                # Extract the numbers from workspace names and find the maximum
-                workspace_numbers = [int(workspace.split('_')[1]) for workspace in existing_workspaces if workspace.startswith('workspace_')]
-                next_workspace_number = max(workspace_numbers) + 1 if workspace_numbers else 1
-            else:
-                next_workspace_number = 1
+        # Handling file upload and dataset creation
+        if 'dname' in request.FILES and 'description' in request.POST and 'repo' in request.POST:
+            # Get the start time from session or calculate it based on form submission
+            start_time = float(request.session.get('uploadStartTime', time.time() * 1000)) / 1000
+            dname = request.FILES.get('dname')
+            description = request.POST['description']
+            repo = request.POST['repo']
+            datastore = request.POST['datastore']
+            datastore_instance = Datastores.objects.get(UUID=datastore)
+            user_instance = User.objects.get(username=username)
             
-            # Create the directory for the new workspace
-            new_workspace_dir = os.path.join(user_workspace_dir, f'workspace_{next_workspace_number}')
-            os.makedirs(new_workspace_dir, exist_ok=True)
-
-            # Save the uploaded file directly into the new workspace directory
-            fs = FileSystemStorage(location=new_workspace_dir)
-            fs.save(uploaded_file.name, uploaded_file)
-            uploaded_file_name = uploaded_file.name
+            manager = getDataStoreManager()
+            datastore = manager.getDatastore(datastore)
+            datasetid = datastore.putDataset(dname)
             
-            # Read the CSV file to extract column names
-            try:
-                with fs.open(uploaded_file.name, 'r') as file:
-                    reader = csv.reader(file)
-                    column_names = next(reader)  # Get the header row
-            except Exception as e:
-                # Handle any errors that might occur during file reading
-                print("Error reading CSV file:", e)
-            # Render upload.html with uploaded file name and column names
-            return render(request, 'upload.html', {'username': username,'uploaded_file_name': uploaded_file_name, 'column_names': column_names})
+            metadata = Datasets(
+                Username=user_instance,
+                Name=dname.name,
+                UUID=datasetid,
+                Datastore_UUID = datastore_instance,
+                Description=description,
+                Repository=repo,
+                Created=timezone.now().date(), 
+            )
+            metadata.save()
```

Comment:
There is no metadata table.

Reply:
metadata is the name of the variable here, the table name is correctly used as Datasets.